src/main/java/aero/t2s/modes/CprPosition.java

```diff
@@ -40,6 +55,41 @@ public double getLon() {
         return lon;
     }
 
+    public void setSurface(boolean surface) {
+        this.surface = surface;
+    }
+
+    public boolean getSurface() {
+        return this.surface;
+    }
+
+    public void validateSurface() {
+        // For SurfacePositions we get 8 possible solutions: 2x latitude, 4x longitude zones at 90 degree increments
+        // TODO Could use receiver/reference lat/lon to chose which possibile solution is closest... or build-up an average over data received to-date
```

Comment:
Closest to the reference point for the first decoding then for the second decoding the reference point should change to the previous position. This will be I think the most complete solution